### PR TITLE
enable gosec

### PR DESCRIPTION
**Pull request by kim-tsao**
Signed-off-by: Kim Tsao <ktsao@redhat.com>

### What does this PR do?:

enables the gosec scanner and addresses any high sev findings

### Which issue(s) this PR fixes:

https://github.com/devfile/api/issues/937

### PR acceptance criteria:
Testing and documentation do not need to be complete in order for this PR to be approved. We just need to ensure tracking issues are opened.

> - Open new test/doc issues under the [devfile/api](https://github.com/devfile/api/issues) repo
> - Check each criteria if:
>  - There is a separate tracking issue. Add the issue link under the criteria
>  **or**
>  - test/doc updates are made as part of this PR
> -  If unchecked, explain why it's not needed

- [ ] Unit/Functional tests

  

- [ ] [QE Integration test](https://github.com/devfile/integration-tests) 

  

- [ ] Documentation 

   

- [ ] Client Impact

### CODE REVIEW

File: pkg/apis/workspaces/v1alpha1/component_plugin_conversion.go [hunk 1]
```diff
@@ -27,6 +27,7 @@ func convertPluginComponentTo_v1alpha2(srcComponent *Component, destComponent *v
 	destComponent.Name = pluginKey
 
 	for _, srcCommand := range src.Commands {
+		srcCommand := srcCommand
```
Comment (amisevsk) [inline pkg/apis/workspaces/v1alpha1/component_plugin_conversion.go:30]: This line appears to be a no-op?

File: run_gosec.sh [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+# This script runs the gosec scanner locally
+
+if ! command -v gosec 2> /dev/null
+then
+  echo "error gosec must be installed with this command: go install github.com/securego/gosec/v2/cmd/gosec@latest" && exit 1
+fi
+
+gosec -no-fail -fmt=sarif -out=gosec.sarif -exclude-dir test  -exclude-dir generator  ./...
```
Comment (amisevsk) [inline run_gosec.sh:9]: Trailling newline missing.

File: pkg/apis/workspaces/v1alpha1/component_plugin_conversion.go (hunk 1 above)
Comment (kim-tsao) [reply pkg/apis/workspaces/v1alpha1/component_plugin_conversion.go:30]: it ensures the address is not reused.  It's one of the approaches you can use to fix this issue (the other is to use an index) according to this discussion: https://stackoverflow.com/questions/62446118/implicit-memory-aliasing-in-for-loop.

Comment (amisevsk) [reply pkg/apis/workspaces/v1alpha1/component_plugin_conversion.go:30]: Ah makes sense, I didn't realize we used `&srcCommand` below.

**Review by amisevsk — APPROVED** [review]

**Pull request merged**